core/src/main/java/org/jboss/sbomer/core/utils/SbomUtils.java

```diff
@@ -88,8 +98,17 @@ public static Optional<Property> findPropertyWithNameInComponent(String property
     }
 
     public static boolean hasExternalReference(Component c, ExternalReference.Type type) {
-        return c.getExternalReferences() != null
-                && c.getExternalReferences().stream().filter(ref -> ref.getType().equals(type)).count() > 0;
+        return getExternalReferences(c, type).size() > 0;
+    }
+
+    public static List<ExternalReference> getExternalReferences(Component c, ExternalReference.Type type) {
+        List<ExternalReference> filteredExternalReferences = Optional.ofNullable(c.getExternalReferences())
+                .map(Collection::stream)
+                .orElseGet(Stream::empty)
+                .filter(ref -> ref.getType().equals(type))
+                .toList();
+
+        return filteredExternalReferences;
     }
 
     public static void addExternalReference(Component c, ExternalReference.Type type, String url, String comment) {
```

Comment:
this was probably not rebased on latest commits.

Reply:
I just checked this one, and it looks exactly like it is in latest `main`. Am I missing something?

Reply:
sorry my bad, looks ok. I was confused by previous comment.